benchmark/custom_csv_formater.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <benchmark/benchmark.h>
+#include <vector>
+namespace benchmark {
+
+class customCSVReporter : public BenchmarkReporter {
+public:
+  customCSVReporter() : printed_header_(false) {}
+  bool ReportContext(const Context &context) override;
+  void ReportRuns(const std::vector<Run> &reports) override;
+
+private:
+  std::string CsvEscape(const std::string &s) {
+    std::string tmp;
+    tmp.reserve(s.size() + 2);
+    for (char c : s) {
+      switch (c) {
+      case '"':
+        tmp += "\"\"";
+        break;
+      default:
+        tmp += c;
+        break;
+      }
+    }
+    return '"' + tmp + '"';
+  }
+
+  // Function to return an string for the calculated complexity
+  std::string GetBigOString(BigO complexity) {
+    switch (complexity) {
+    case oN:
+      return "N";
+    case oNSquared:
+      return "N^2";
+    case oNCubed:
+      return "N^3";
+    case oLogN:
+      return "lgN";
+    case oNLogN:
+      return "NlgN";
+    case o1:
+      return "(1)";
+    default:
+      return "f(N)";
+    }
+  }
+
+  void PrintRunData(const Run &report);
+  bool printed_header_;
+  std::set<std::string> user_counter_names_;
+
+  std::ostream *nullLog = nullptr;
+
+  std::vector<std::string> elements = {
```

Comment:
Just a suggestion, but since this vector doesn't change, it might be a nice to use std::array instead and make it const, eg:
```
#include <array>
const std::array<const std::string, 12> elements = { ... };
```